unstructured/ingest/connector/slack.py

```diff
@@ -0,0 +1,226 @@
+
+
+@dataclass
+class SlackIngestDoc(BaseIngestDoc):
+    """Class encapsulating fetching a doc and writing processed results (but not
+    doing the processing!).
+
+    Also includes a cleanup method. When things go wrong and the cleanup
+    method is not called, the file is left behind on the filesystem to assist debugging.
+    """
+
+    config: SimpleSlackConfig
+    channel: str
+    token: str
+    oldest: str
+    latest: str
+
+    # NOTE(crag): probably doesn't matter,  but intentionally not defining tmp_download_file
+    # __post_init__ for multiprocessing simplicity (no Path objects in initially
+    # instantiated object)
+    def _tmp_download_file(self):
+        channel_file = self.channel + ".txt"
+        return Path(self.config.download_dir) / channel_file
+
+    def _output_filename(self):
+        output_file = self.channel + ".json"
+        return Path(self.config.output_dir) / output_file
+
+    def has_output(self):
+        """Determine if structured output for this doc already exists."""
+        return self._output_filename().is_file() and os.path.getsize(self._output_filename())
+
+    def _create_full_tmp_dir_path(self):
+        self._tmp_download_file().parent.mkdir(parents=True, exist_ok=True)
+
+    @requires_dependencies(dependencies=["slack_sdk"], extras="slack")
+    def get_file(self):
+        """Fetches the data from a slack channel and stores it locally."""
+
+        self._create_full_tmp_dir_path()
+        if (
+            not self.config.re_download
+            and self._tmp_download_file().is_file()
+            and os.path.getsize(self._tmp_download_file())
+        ):
+            if self.config.verbose:
+                logger.debug(f"File exists: {self._tmp_download_file()}, skipping download")
+            return
+
+        if self.config.verbose:
+            logger.debug(f"fetching channel {self.channel} - PID: {os.getpid()}")
+
+        messages = []
+        self.client = WebClient(token=self.token)
+
+        try:
+            oldest = "0"
+            latest = "0"
+            if self.oldest:
+                oldest = self.convert_datetime(self.oldest)
+
+            if self.latest:
+                latest = self.convert_datetime(self.latest)
+
+            result = self.client.conversations_history(
+                channel=self.channel,
+                oldest=oldest,
+                latest=latest,
+            )
+            messages.extend(result["messages"])
+            while result["has_more"]:
+                result = self.client.conversations_history(
+                    channel=self.channel,
+                    oldest=oldest,
+                    latest=latest,
+                    cursor=result["response_metadata"]["next_cursor"],
+                )
+                messages.extend(result["messages"])
+        except SlackApiError as e:
+            logger.error(f"Error: {e}")
+
+        with open(self._tmp_download_file(), "w") as f:
```

Comment:
total nit, but something a bit more verbose than f for the file handle would be preferred